lint/src/main/java/org/datacommons/lint/Lint.java

```diff
@@ -0,0 +1,162 @@
+package org.datacommons.lint;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.Callable;
+import org.datacommons.proto.Debug;
+import org.datacommons.proto.Mcf;
+import org.datacommons.util.McfParser;
+import org.datacommons.util.McfUtil;
+import org.datacommons.util.TmcfCsvParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
+
+@CommandLine.Command(
+    name = "DCLint",
+    mixinStandardHelpOptions = true,
+    version = "DCLint 0.1",
+    description = "Tool for use in developing datasets for Data Commons.")
+class Lint implements Callable<Integer> {
+  private static final Logger logger = LoggerFactory.getLogger(Lint.class);
+
+  @CommandLine.Parameters(
+      arity = "1..*",
+      description =
+          ("List of input files. This depends on the --format option. "
+              + "For 'tmcf', the files are expected to be "
+              + "Template MCF files. For 'mcf', the files are expected to be Instance MCF "
+              + "files. For tmcfCsv, the first file should be a Template MCF file, followed "
+              + "by one or more CSVs compatible with the provided TMCF file."))
+  private File[] files;
+
+  enum FormatType {
+    mcf,
+    tmcf,
+    tmcfCsv
+  };
+
+  @CommandLine.Option(
+      names = {"-f", "--format"},
+      description = "Format of input files: ${COMPLETION-CANDIDATES}",
+      required = true)
+  private FormatType formatType;
+
+  enum CommandType {
+    chk,
+    genMcf
+  };
+
+  @CommandLine.Option(
+      names = {"-c", "--cmd"},
+      description =
+          "The command to run: ${COMPLETION-CANDIDATES}.  'chk' runs "
+              + "syntax checks on the input files.  'genMcf' produces instance MCF output, "
+              + "and only makes sense when --format is 'tmcfCsv'. Default: 'chk'",
+      defaultValue = "chk",
+      required = true)
+  private CommandType cmdType;
+
+  @CommandLine.Option(
+      names = {"-d", "--delimiter"},
```

Comment:
also from a "linter" point of view, should the tool determine the delimiter rather than let user specify it?

Reply:
Changed to use better default for .csv vs. .tsv. We can parse/interpret it, maybe in follow-on. I do wonder about  something like though:

`This,Should|Be,What|Delimiter`

Or, we should at least restrict characters that can go into the header, or something.  Or, provide feedback to user that this is how we interpreted the delimiter.  I just feel this one option is simple enough to provide for starters?

Reply:
I see, a default delimiter helps.